airflow/www/widgets.py

```diff
@@ -40,6 +40,9 @@ class AirflowDateTimePickerWidget:
         "</div>"
     )
 
+    def __init__(self, input_required: bool = True):
```

Comment:
@xuganyu96 Thanks for the contribution! :)

I like the idea of making datetime picker optional, but I foresee an issue, I have not gone through the code, but there can be multiple ways you can enforce an input field to be required, adding the `required` attribute is one of them. So I think adding just this code won't guarantee the behavior for all occurrences. It would be much nicer if we can see in a single PR the change and its usage.

Reply:
Thank you for the feedback.

I am not entirely familiar with using JS to control attributes of an HTML tag. My intended use case is "outside" the Airflow codebase, such as with developing plugins. For relatively simple plugins it would be great to use widgets with Flask/Jinja2 and not have to work with JS, which is why I want to parameterize the `required` attribute with Python instead of anywhere else.